applications/GeoMechanicsApplication/tests/cpp_tests/test_process_factory.cpp

```diff
@@ -0,0 +1,123 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Wijtze Pieter Kikstra
+//                   Anne van de Graaf
+//
+
+#include "testing/testing.h"
+#include "custom_utilities/process_factory.hpp"
+
+using namespace Kratos;
+
+
+namespace
+{
+
+class FooProcess : public Process
+{
+};
+
+class BarProcess : public Process
+{
+};
+
+}
+
+
+namespace Kratos::Testing
+{
+
+KRATOS_TEST_CASE_IN_SUITE(CreateNothingWhenNoCreatorWasAddedForRequestedProcess, KratosGeoMechanicsFastSuite)
+{
+    ProcessFactory factory;
+
+    const Parameters process_settings;
+    const auto process = factory.Create("UnknownProcess", process_settings);
+
+    KRATOS_EXPECT_FALSE(process)
+}
+
+KRATOS_TEST_CASE_IN_SUITE(CreateNothingWhenTheAddedCreatorIsEmpty, KratosGeoMechanicsFastSuite)
+{
+    ProcessFactory factory;
+    factory.AddCreator("TestProcess", {});
+
+    const Parameters process_settings;
+    const auto process = factory.Create("TestProcess", process_settings);
+
+    KRATOS_EXPECT_FALSE(process)
+}
+
+KRATOS_TEST_CASE_IN_SUITE(CreateFooProcessAfterCreatorWasAdded, KratosGeoMechanicsFastSuite)
+{
+    ProcessFactory factory;
+    factory.AddCreator("FooProcess", [](const Parameters&){return std::make_unique<FooProcess>();});
+
+    const Parameters process_settings;
+    const auto process = factory.Create("FooProcess", process_settings);
+
+    KRATOS_EXPECT_TRUE(dynamic_cast<FooProcess*>(process.get()))
+    KRATOS_EXPECT_FALSE(dynamic_cast<BarProcess*>(process.get()))
+}
+
+KRATOS_TEST_CASE_IN_SUITE(CreateDifferentKindsOfProcessesAfterAddingCreators, KratosGeoMechanicsFastSuite)
+{
+    ProcessFactory factory;
+    factory.AddCreator("FooProcess", [](const Parameters&){return std::make_unique<FooProcess>();});
+    factory.AddCreator("BarProcess", [](const Parameters&){return std::make_unique<BarProcess>();});
+
+    const Parameters foo_process_settings;
+    const auto foo_process = factory.Create("FooProcess", foo_process_settings);
+    const Parameters bar_process_settings;
+    const auto bar_process = factory.Create("BarProcess", bar_process_settings);
+
+    KRATOS_EXPECT_TRUE(dynamic_cast<FooProcess*>(foo_process.get()))
+    KRATOS_EXPECT_TRUE(dynamic_cast<BarProcess*>(bar_process.get()))
+}
+
+KRATOS_TEST_CASE_IN_SUITE(CreatorReceivesGivenSettingsWhenMakingNewProcess, KratosGeoMechanicsFastSuite)
+{
+    ProcessFactory factory;
+    const Parameters* received_settings = nullptr;
```

Comment:
In my opinion, this const is a bit misleading, since the pointer itself is changed later on. It's fine like this as well (especially since the scope of the test is nice and small).

Reply:
My intention is to express that I just want to observe the object pointed to, not to modify it through this pointer. I feel that every C++ programmer should understand the difference between `const X*`, `X* const` and `const X* const`.

Reply:
Agreed